src/frequenz/client/base/channel.py

```diff
@@ -41,6 +42,24 @@ class SslOptions:
     """
 
 
+@dataclasses.dataclass(frozen=True)
+class KeepAliveOptions:
+    """Options for HTTP2 keep-alive pings."""
+
+    enabled: bool = True
+    """Whether keep-alive should be enabled."""
+
+    interval: timedelta | None = None
+    """The interval between pings.
+
+    If None, keep-alive is disabled."""
```

Comment:
I would not allow `None` here, and provide some default, as `enabled` already covers for the disabled case. Otherwise having `enabled=True, interval=None` doesn't make much sense (also the current default is weird, it is `enabled=True` but it is not actually enabled.

For example:
```suggestion
    interval: timedelta = timedelta(seconds=5)
    """The interval between pings."""
```

Reply:
There may not be a default keep-alive that works for the general case, but 5s seems excessive.  Nginx waits for 75s, so I'll set it to 60s: http://nginx.org/en/docs/http/ngx_http_core_module.html#keepalive_timeout

Reply:
> also the current default is weird, it is `enabled=True` but it is not actually enabled.

:D true, I didn't think of that.

Reply:
Just for reference, Go sets **TCP** keepalive to 15 seconds intervals, [this article](https://www.evanjones.ca/tcp-connection-timeouts.html) does a small analysis about what it means for data transmission:

>The disadvantage is this sends additional keepalive TCP packets for idle connections, but 1 packet every 15 seconds sounds reasonable to me ([the Go team agrees](https://github.com/golang/go/issues/23459); 100k connections = 6.7k packets/sec = 417 kiB/sec = 0.3% of a 1 Gbps Ethernet).

http/2 keepalive probably have quite a bit more overhead, I'm just putting it here for some context and to remember that this is also for detecting broken connections. Probably for could services 1 minute is enough and probably that's most of the services that we'll have, so 60s seems a reasonable default I guess.

As for the "avoid the connection to be dropped because of inactivity" and also for reference, here are some examples of network actors that could drop idle connections:

* [Amazon network load balancers](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/network-load-balancers.html#connection-idle-timeout): After ~6 minutes (350 seconds, configurable). TCP keep-alive is considered network activity, so it can be used to avoid the connection drop.
* [Azure load balancers](https://learn.microsoft.com/en-us/azure/load-balancer/load-balancer-tcp-reset#configurable-tcp-idle-timeout): After 4 minutes (configurable from 4 to 100 minutes). TCP keep-alive is also considered network activity, so it can be used to avoid the connection drop.
* Google ignore http/2 keepalive, so for this use case we need to send an application-level packet to keep connections alive.

Reply:
done